docker/centos7.3/Dockerfile

```diff
@@ -0,0 +1,148 @@
+# Setup access
+#################################################################################
+RUN mkdir -p /tmp/image-build-space/access
+
+COPY docker/centos7.3/files/access/* /tmp/image-build-space/access/
+
+COPY docker/centos7.3/files/pam/* /tmp/image-build-space/access/
+
+COPY docker/centos7.3/files/image-prep-scripts/setup-access.sh /tmp/image-build-space/image-prep-scripts
+RUN /tmp/image-build-space/image-prep-scripts/setup-access.sh
+
+COPY docker/centos7.3/files/image-prep-scripts/hwx-internal-setup.sh /tmp/image-build-space/image-prep-scripts
+RUN /tmp/image-build-space/image-prep-scripts/hwx-internal-setup.sh
+#################################################################################
+
+# install iproute explicitly on centos7, it's present by default in centos6, debian7
+RUN yum -y install iproute
+
+#################################################################################
+# Setup access
+#################################################################################
+RUN mkdir -p /bootstrap /bootstrap/logs
+
+COPY docker/centos7.3/files/image-prep-scripts/privileged-start-services-script.c /tmp/image-build-space/image-prep-scripts/
+
+RUN gcc /tmp/image-build-space/image-prep-scripts/privileged-start-services-script.c -D COMMAND=\"/bootstrap/start-services-script.sh\" -o /bootstrap/start-systemd
+RUN chgrp -v nobody /bootstrap/start-systemd
+RUN chmod 4050 /bootstrap/start-systemd
+
+ADD docker/centos7.3/files/image-runtime-scripts/start-services-script.sh /bootstrap/
+
+#################################################################################
+# Operating system specific stuff
+#################################################################################
+COPY docker/centos7.3/files/base-centos7.3-os-specific-build.sh /tmp/image-build-space/
+RUN /tmp/image-build-space/base-centos7.3-os-specific-build.sh
+
+# Get rid "Unable to get valid context .." message
+COPY docker/centos7.3/files/autorelabel /.autorelabel
+
+#################################################################################
+# Install Cloudbreak dependencies
+#################################################################################
+# Use docker-systemctl-replacement during SaltStack provisioning:
+# `systemctl start xyz` commands are executed during highstate, which is not supported
+# by systemd. docker-systemctl-replacement overcomes this limitation.
+
+# Need to explicitly call yum update before linking systemctl-replacement to avoid
+# the later update (executed through Salt recipe) overwriting our fix.
+RUN yum -y update
+ADD ${SYSTEMCTL_REPLACEMENT} /usr/bin
+RUN chmod 755 /usr/bin/$(basename ${SYSTEMCTL_REPLACEMENT}) && \
+    mv /usr/bin/systemctl /usr/bin/systemctl.orig && \
+    ln -sf /usr/bin/systemctl.py /usr/bin/systemctl
+
+ADD /saltstack /tmp/saltstack
+ADD ${SALT_REPO_URL} /tmp/saltstack/
+ADD /scripts/salt-install.sh /tmp/saltstack/
+ADD /scripts/salt-setup.sh /tmp/saltstack/
```

Comment:
Use COPY instead of ADD where appropriate
https://docs.docker.com/engine/userguide/eng-image/dockerfile_best-practices/#add-or-copy